CTFd/utils/email/__init__.py

```diff
@@ -139,6 +139,18 @@ def check_email_is_whitelisted(email_address):
     domain_whitelist = get_config("domain_whitelist")
     if domain_whitelist:
         domain_whitelist = [d.strip() for d in domain_whitelist.split(",")]
-        if domain not in domain_whitelist:
-            return False
+
```

Comment:
Originally I thought that this code didn't work for `*.example.com` but I believe it does. The user would just need to remember that `*.example.com` does not also include `example.com` so they would need to add both. 

Also this logic does allow things like `*ample.com` which I think is okay.

Reply:
It's all in the handling of the suffix - we can either strip out just the asterisk:

`suffix = allowed_domain[1:]` which requires the user to set both `example.com, *.example.com`

or alternatively, also strip the dot `*.`:

`suffix = allowed_domain[2:]` which then allows things like `john.doe@foobarexample.com`

I think from these options I actually prefer the first case.

Reply:
We should allow `*.example.com` but not something like `*ample.com`. We don't want someone to rely on the `*ample.com` behavior for some reason and then we need to change it. 

So the overall logic here is right. Now my question is what to do about the `* in domain` check.

Reply:
I get the sense that the * in domain check is fine as it will prevent sending out invalid confirmation emails since * isn't valid in a domain. I am good with this PR now.